### PR TITLE
Replacing deprecated softmax_cross_entropy_with_logits function

### PR DESCRIPTION
**Pull request by grigala**
Future major versions of TensorFlow will allow gradients to flow into the labels input on backprop by default.

[`tf.nn.softmax_cross_entropy_with_logits`](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits) (from [`tensorflow.python.ops.nn_ops`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py)) is deprecated and will be removed in a future version.

Replacing it with [`tf.nn.softmax_cross_entropy_with_logits_v2`](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits_v2)

### CODE REVIEW

**Comment by martin-gorner** [comment]
Thank you for your contribution. I am leaving it as it is for the time being. The warning is explicit enough, the migration path clear. However, the current code works and is simpler.
